packages/wethegit-components/src/hooks/use-breakpoints/use-breakpoints.ts

```diff
@@ -0,0 +1,37 @@
+import { useEffect, useState } from "react"
+
+import breakpointSettings from "./use-breakpoints.module.scss"
+
+const BREAKPOINTS: [Breakpoint, string][] = Object.entries(breakpointSettings).map(
+  ([key, val]) => [key as Breakpoint, val.replaceAll('"', "")]
+)
+
+export function useBreakpoints() {
+  const [breakpoint, setBreakpoint] = useState<Breakpoint | undefined>("sm")
+
+  useEffect(() => {
+    let resizeTimer: ReturnType<typeof setTimeout>
+
+    function onResize() {
+      clearTimeout(resizeTimer)
+      resizeTimer = setTimeout(() => {
+        setBreakpoint(BREAKPOINTS.find((bp) => window.matchMedia(bp[1]).matches)?.[0])
+      }, 200)
+    }
+
+    onResize()
+
+    window.addEventListener("resize", onResize)
+
+    return () => {
+      clearTimeout(resizeTimer)
+      window.removeEventListener("resize", onResize)
+    }
+  }, [])
+
+  return {
+    breakpoint,
+    mdUp: breakpoint && breakpoint !== "sm",
+    lgUp: breakpoint && !["sm", "md"].includes(breakpoint),
+  }
```

Comment:
```suggestion
  let xxlUp, xlUp, lgUp, mdUp;
  if (breakpoint) {
    xxlUp = breakpoint === 'xxl'
    xlUp = xxlUp || breakpoint === 'xl'
    lgUp = xlUp || breakpoint === 'lg'
    mdUp = lgUp || breakpoint === 'md'
  }
  
  return {
    breakpoint,
    mdUp,
    lgUp,
    xlUp,
    xxlUp,
  }
```


Hey @andrewrubin what do you think of this?

Reply:
I agree, we might as well do it for everything. I only ever use mediumUp and largeUp but it's a good idea to include them!

Reply:
Added. Just for proof that it works:


https://github.com/wethegit/component-library/assets/30575213/3cf371cf-f2ac-4aac-96c8-452322066283